mm2src/coins/eth/wallet_connect.rs

```diff
@@ -0,0 +1,267 @@
+
+        Ok(chain_id)
+    }
+
+    async fn wc_sign_tx<'a>(
+        &self,
+        wc: &WalletConnectCtx,
+        params: Self::Params<'a>,
+    ) -> Result<Self::SignTxData, Self::Error> {
+        let bytes = {
+            let chain_id = self.wc_chain_id(wc).await?;
+            let tx_json = params.prepare_wc_tx_format()?;
+            let session_topic = self.session_topic()?;
+            let tx_hex: String = wc
+                .send_session_request_and_wait(session_topic, &chain_id, WcRequestMethods::EthSignTransaction, tx_json)
+                .await?;
+            // First 4 bytes from WalletConnect represents Protoc info
+            hex::decode(&tx_hex[4..])?
+        };
+        let unverified = rlp::decode(&bytes)?;
+        let signed = SignedTransaction::new(unverified)?;
+        let bytes = rlp::encode(&signed);
+
+        Ok((signed, BytesJson::from(bytes.to_vec())))
+    }
+
+    async fn wc_send_tx<'a>(
+        &self,
+        wc: &WalletConnectCtx,
+        params: Self::Params<'a>,
+    ) -> Result<Self::SignTxData, Self::Error> {
+        let tx_hash: String = {
+            let chain_id = self.wc_chain_id(wc).await?;
+            let tx_json = params.prepare_wc_tx_format()?;
+            let session_topic = self.session_topic()?;
+            wc.send_session_request_and_wait(session_topic, &chain_id, WcRequestMethods::EthSendTransaction, tx_json)
+                .await?
+        };
+
+        let tx_hash = tx_hash.strip_prefix("0x").unwrap_or(&tx_hash);
+        let maybe_signed_tx = {
+            self.wait_for_tx_appears_on_rpc(H256::from_slice(&hex::decode(tx_hash)?), WAIT_RPC_TIMEOUT_SECS, 1.)
+                .await
+                .mm_err(|err| EthWalletConnectError::InternalError(err.to_string()))?
+        };
+        let signed_tx = maybe_signed_tx.ok_or(MmError::new(EthWalletConnectError::InternalError(format!(
+            "Waited too long until the transaction {tx_hash:?} appear on the RPC node"
+        ))))?;
+        let tx_hex = BytesJson::from(rlp::encode(&signed_tx).to_vec());
+
+        Ok((signed_tx, tx_hex))
+    }
+
+    fn session_topic(&self) -> Result<&str, Self::Error> {
+        if let EthPrivKeyPolicy::WalletConnect { ref session_topic, .. } = &self.priv_key_policy {
+            return Ok(session_topic);
+        }
+
+        MmError::err(EthWalletConnectError::SessionError(format!(
+            "{} is not activated via WalletConnect",
+            self.ticker()
+        )))
+    }
+}
+
+pub async fn eth_request_wc_personal_sign(
+    wc: &WalletConnectCtx,
+    session_topic: &str,
+    chain_id: u64,
+) -> MmResult<(H520, Address), EthWalletConnectError> {
+    let chain_id = WcChainId::new_eip155(chain_id.to_string());
+    wc.validate_update_active_chain_id(session_topic, &chain_id).await?;
+
+    let (account_str, _) = wc.get_account_and_properties_for_chain_id(session_topic, &chain_id)?;
+    let message = "Authenticate with Komodefi";
```

Comment:
Is this going to show-up on the external wallet screen?

Reply:
Yes

Reply:
I think this needs approval from various people outside our team whether we want to present ourselves as `Komodefi` or we should prefer some other name.

cc @ca333